core/Translation/Translator.php

```diff
@@ -90,6 +104,67 @@ public function translate($translationId, $args = array(), $language = null)
     }
 
     /**
+     * Returns the country name for the given country code translated to the given language
+     *
+     * @param string $countryCode ISO 3166-1 alpha-2 country code
+     * @param null|string $language ISO 639-1 language code   (defaults to current language)
+     * @return string  translated country name
+     */
+    public function getTranslatedCountry($countryCode, $language=null)
+    {
+        $filePattern = PIWIK_INCLUDE_PATH . '/vendor/piwik/i18n-data/countries/%s.json';
+
+        $language = is_string($language) ? $language : $this->getCurrentLanguage();
+        $countryCode = strtoupper($countryCode);
+
+        // try to get country name in given language
+        if (!array_key_exists($language, $this->loadedCountryTranslations) && file_exists(sprintf($filePattern, $language))) {
+            $this->loadedCountryTranslations[$language] = @json_decode(file_get_contents(sprintf($filePattern, $language)), true);
```

Comment:
Should we maybe throw an exception if `json_decode` returns null meaning it fails? Maybe it would allow us to detect possible errors in a JSON file faster. But not really important I reckon.